client/dashboard/app/queries/me-profile.ts

```diff
@@ -1,17 +1,20 @@
+import { queryOptions, mutationOptions } from '@tanstack/react-query';
 import { fetchProfile, updateProfile } from '../../data/me-profile';
 import { queryClient } from '../query-client';
-import type { UserProfile } from '../../data/me-profile';
 
-export const profileQuery = () => ( {
-	queryKey: [ 'me', 'profile' ],
-	queryFn: fetchProfile,
-} );
+export const profileQuery = () =>
+	queryOptions( {
+		queryKey: [ 'me', 'profile' ],
+		queryFn: fetchProfile,
+	} );
 
-export const profileMutation = () => ( {
-	mutationFn: updateProfile,
-	onSuccess: ( newData: Partial< UserProfile > ) => {
-		queryClient.setQueryData( profileQuery().queryKey, ( oldData: UserProfile | undefined ) =>
-			oldData ? { ...oldData, ...newData } : newData
-		);
-	},
-} );
+export const profileMutation = () =>
+	mutationOptions( {
+		mutationFn: updateProfile,
+		onSuccess: ( newData ) => {
+			queryClient.setQueryData(
+				profileQuery().queryKey,
+				( oldData ) => oldData && { ...oldData, ...newData }
```

Comment:
Fixes a bug.
Here, if the profile got mutated before they had ever been queried, then this could would have put a partial profile object into the query cache, when TypeScript thinks it would have been a complete profile object. `"value is undefined"` bugs averted!

To fix the issue, I decided that if the profile gets mutated before ever queried, then we just won't put anything into the query cache.